### ISSUE TITLE
I want to do semantic segmentation on unlabeled point cloud data (xyzrgb) using a pre-trained model.

### ISSUE THREAD

**Issue by Dsai1006** [HuguesTHOMAS/KPConv#91]
Hello. 

I want to do semantic segmentation on unlabeled point cloud data (xyzrgb) using a pre-trained model.
I will use a pre-trained model using S3DIS. What is input format, rgbxyz?
I would like to know how to do semantic segmentation on my own data.

Thank you.

**Comment by HuguesTHOMAS** [comment]
Hi @Dsai1006,

To test the network on your own data, you will have to create as new dataset class by copying `datasets/S3DIS` and modifying everything that need to be. More details are explained [here](https://github.com/HuguesTHOMAS/KPConv/blob/master/doc/new_dataset_guide.md). You can try the `test_any_model.py` script on a dataset like Semantic3D which has an unlabeled test set to see which functions are called and follow the flow of computations.

To answer your question about the input format, the points are considerated separately so the input points will be xyz and the input features will be 1rgb or 1rgbz if you feel the height could help the segmentation. You can find more details on this input format in the paper.

**Comment by lwx-bupt** [comment]
Hello.
Same as @Dsai1006 , I'm also doing such a test. I 'm trying it on a single *.ply file which contains XYZ coordinates in float number and RGB color for each point, but the forwarding process doesn't seem to be running properly. So what do I need to do in order to see what has gone wrong?

**Comment by HuguesTHOMAS** [comment]
Hi @lwx-bupt,

If you use a network that was trained on S3DIS, it should be fine. You can look at the `load_subsampled_cloud` method in `datasets/S3DIS`. This is where your data will be loaded and processed. Try to print the shapes and dtypes of the arrays that are loaded and verify that everything is going well there.

Best,
Hugues

**Comment by lwx-bupt** [comment]
Thanks a lot, and now I've finished checking that part. However another issue arises which says "failed to run cuBLAS routine cublasGemmBatchedEx: CUBLAS_STATUS_NOT_SUPPORTED". Is it caused by cuda version?

**Comment by lwx-bupt** [comment]
Here's my Nvidia-smi log
Sun Jul 12 09:54:59 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.100      Driver Version: 440.100      CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Quadro P600         Off  | 00000000:01:00.0 Off |                  N/A |
| N/A   52C    P5    N/A /  N/A |    426MiB /  4040MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

**Comment by lwx-bupt** [comment]
I'm running it in tf-gpu 1.12, cudnn 7.6.5 and cudatoolkit 9.2 in my (conda) environment. Should I upgrade my cudatoolkit(in conda)?

**Comment by lwx-bupt** [comment]
Python console log here:

**Comment by HuguesTHOMAS** [comment]
It seems you have a problem with your CUDA/Tensorflow instalation. I cannot really help you with that. Try to figure out if a simple tf matmul works on GPUon a simple script to debug this.

**Comment by lwx-bupt** [comment]
> It seems you have a problem with your CUDA/Tensorflow instalation. I cannot really help you with that. Try to figure out if a simple tf matmul works on GPUon a simple script to debug this.

Thanks. Fixed it by reinstalling tf with pip~
